### PR TITLE
BUG: Fix indexed/CMYK images

### LINKED ISSUE #2030: Image extraction raises "ValueError: illegal image mode"
I am trying to extract the images from a PDF file, which fails with "illegal image mode"

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
Linux-5.14.21-150400.24.69-default-x86_64-with-glibc2.31

$ python -c "import pypdf;print(pypdf.__version__)"
3.13.0
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
from pypdf import PdfReader

reader = PdfReader('redacted.pdf')
for page in reader.pages:
    for image in reader.images:
        print(image)
```

I have generated a mostly empty version of the document reproducing this issue: [redacted.pdf](https://github.com/py-pdf/pypdf/files/12182021/redacted.pdf)

## Traceback

This is the complete traceback I see:

```
Traceback (most recent call last):
  File "/home/stefan/temp/analyze.py", line 17, in <module>
    for image in page.images:
  File "/home/stefan/temp/pypdf/pypdf/_page.py", line 2604, in __iter__
    yield self[i]
  File "/home/stefan/temp/pypdf/pypdf/_page.py", line 2600, in __getitem__
    return self.get_function(lst[index])
  File "/home/stefan/temp/pypdf/pypdf/_page.py", line 522, in _get_image
    imgd = _xobj_to_image(cast(DictionaryObject, xobjs[id]))
  File "/home/stefan/temp/pypdf/pypdf/filters.py", line 892, in _xobj_to_image
    img, image_format, extension = _handle_flate(
  File "/home/stefan/temp/pypdf/pypdf/filters.py", line 804, in _handle_flate
    img.putpalette(lookup.get_data())
  File "/home/stefan/temp/venv/lib64/python3.9/site-packages/PIL/Image.py", line 1914, in putpalette
    raise ValueError("illegal image mode")
ValueError: illegal image mode
```

### PR DESCRIPTION
**Pull request by pubpub-zz**
closes #2030

there is a TODO in pypdf/filters.py:L889 as currently pillow does not accept( cf CMYK palette in https://github.com/python-pillow/Pillow/issues/7309)

### CODE REVIEW

File: pypdf/filters.py [hunk 1]
```diff
@@ -865,11 +886,22 @@ def bits2byte(data: bytes, size: Tuple[int, int], bits: int) -> bytes:
                         # gray lookup does not work : it is converted to a similar RGB lookup
                         lookup = b"".join([bytes([b, b, b]) for b in lookup])
                         mode = "RGB"
+                    # TODO : this is a work around until PIL is able to process CMYK images
```
Comment (MartinThoma) [inline pypdf/filters.py:889]:
```suggestion
                    # The following is a work around until PIL is able to process CMYK images:
                    # https://github.com/python-pillow/Pillow/issues/7309
```

Comment (pubpub-zz) [reply pypdf/filters.py:889]: Can you keep this TODO to keep some track where to do the upgrade when pillow will be upgraded

Comment (MartinThoma) [reply pypdf/filters.py:889]: I've added the TODO back, but I would like to have the URL of the issue in there. This way it's clearer to readers when this TODO wil dissappear (although one of us will very likely handle it :smile: )

Comment (pubpub-zz) [reply pypdf/filters.py:889]: call me one ??? 🤣🤣🤣

**Pull request merged**